### ISSUE TITLE
Create groups for pages

### ISSUE THREAD

**Issue by sibille** [Microsoft/WindowsTemplateStudio#2338]
We're preparing three ink pages and also page number is growing. I think we should group pages as we do with features. 
It would also be great if page/feature groups could be collapsible/expandible to provide a better overview.

Actions:
- [ ] Define page groups
- [ ] Design a way to collapse/expand groups

**Comment by mrlacey** [comment]
Has anyone reported or provided feedback that the number is too large, hard to navigate, or hard to find what people are looking for?

What benefit would collapsing/expanding groups have?
I presume that they'd all be open by default. Is this the intention?
Do we have any feedback or telemetry that this would be useful?

There was an issue for search previously (#282) but this was closed with a note that it could be opened if _needed_. Has that need now been met/justified?

**Comment by sibille** [comment]
We talked about collapse/expand during the build demos as the feature page is very long. Grouping in pages would be good to have the ink pages together.

**Comment by crutkas** [comment]
I do think this is a good idea but it is lower priority based on our roadmap

**Comment by mrlacey** [comment]
> Grouping in pages would be good to have the ink pages together.

Does not having them next to each other in the list do the same thing?

> We talked about collapse/expand during the build demos as the feature page is very long.

Some because it was discussed it gets done? Who discussed it? What else was discussed?


These arguments sound like they're trying to justify a decision to do some work because it can be done, not because it should be done.

What is the goal here?
What are the alternatives?
Why is what's proposed the best option?

**Comment by sibille** [comment]
I agree, it's not high priority but a nice to have in the wizard. 

If we agree on the basic idea of having groups in pages and features + making them collapsible/expandable we could do some design proposals, discuss them and implement during the summer (if there's time). 
If there are fundamental doubts to grouping pages/features I'll close this issue and open another to discuss alternatives.

**Comment by mrlacey** [comment]
> If we agree on the basic idea of having groups in pages 

You've still not answered why to add groups to pages, other than to have the new Ink pages together, which can be done without groups anyway.

> we could do some design proposals, discuss them and implement 

Always!
It should never be a case of just implementing something. Justify the change, discuss/document options and then implement. 
For something visual, with external validation required, this is is especially important. Implementing something before verifying it risks wasting time as rework takes longer.
Without a justification for making the change, it's also impossible to confirm that the change that's made is appropriate or correct.

**Comment by sibille** [comment]
Got the message ;) I opened this issue to talk about how to group pages and to propose design options. 
If everyone is ok with the general idea we'll propose a design to be able to implement during the summer. 

Specs so far: 
- Add groups to "Add pages" page
- Make groups collapsible/expandable
- All groups are expanded by default
- Add buttons to collapse/expand all groups

Everybody ok with that?

**Comment by mrlacey** [comment]
This

> Got the message 

does not match up with this

> Add groups to "Add pages" page


My point is to define **WHY**  something should be done before doing anything else. What is the point of adding groups to pages?
- What is the problem that adding groups will solve?

- What is the problem that making groups collapsible/expandable will solve?
- What are the alternatives?

**Comment by sibille** [comment]
True, I skipped the why part, sorry. To have the ink pages together in it's own group and provide a better overview as the number of pages is growing, we now have 12, with ink there will be 15. Alternatives: I don't see any right now, but am open to proposals. 
Also, this is not high priority, if we have doubts about this being useful, we can move it to the backlog or close this and review at a later point.
